listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -666,6 +677,155 @@ def dump_listens(self, location, dump_id, start_time=datetime.utcfromtimestamp(0
         self.log.info('Dump present at %s!', archive_path)
         return archive_path
 
+    def write_parquet_files(self, archive_dir, temp_dir, tar_file, start_time=None, end_time=None, full_dump=True):
+        """
+            Carry out fetching listens from the DB, joining them to the MBID mapping table and
+            then writing them to parquet files.
+
+        Args:
+            archive_dir: the directory where the listens dump archive should be created
+            tmp_dir: the directory where tmp files should be written
+            dump_id (int): the ID of the dump in the dump sequence
+            start_time and end_time (datetime): the time range for which listens should be dumped
+                start_time defaults to utc 0 (meaning a full dump) and end_time defaults to the current time
+            full_dump (bool): Is this a full or incremental dump?
+        """
+
+        listen_count = 0
+
+        if start_time:
+            start_time = datetime.utcfromtimestamp(datetime.timestamp(start_time))
+        if end_time:
+            end_time = datetime.utcfromtimestamp(datetime.timestamp(end_time))
+        else:
+            end_time = datetime.now()
+
+        parquet_file_id = 0
+
+        query = """SELECT listened_at,
+                          user_name,
+                          data->'track_metadata'->>'artist_name' AS artist_name,
+                          artist_credit_id,
+                          data->'track_metadata'->>'release_name' AS release_name,
+                          release_mbid::TEXT,
+                          track_name AS recording_name,
+                          recording_mbid::TEXT,
+                          created
+                     FROM listen l
+                     JOIN listen_mbid_mapping m
+                       ON (data->'track_metadata'->'additional_info'->>'recording_msid')::uuid = recording_msid
+                    WHERE created > %s
+                      AND created <= %s
+                      AND recording_mbid IS NOT NULL
+                 ORDER BY created ASC"""
+
+        args = (start_time, end_time)
+
+        listen_count = 0
+
+        current_created = None
+        conn = timescale.engine.raw_connection()
+        with conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor) as curs:
```

Comment:
Now that we are using Pandas, we might as well use its `read_sql`(https://pandas.pydata.org/docs/reference/api/pandas.read_sql.html#pandas-read-sql) utility. The only issue I see with this is how to estimate sizes if we do this.

Reply:
Speaks for keeping it as is, doesn't it. :)

Reply:
Yeah but I think if we read directly into dataframe and then manipulate it instead of doing in it a loop ourself it'll probably be faster.

Reply:
Let continue as is for now and keep in mind that we can use this to speed up the dumps at some point in the future. Like I said, we haven't done a full dump yet, so no idea how long this takes. Faster than before, I'm nearly certain....